mlrun/runtimes/pod.py

```diff
@@ -133,9 +133,21 @@ def to_dict(self, fields=None, exclude=None, strip=False):
     def apply(self, modify):
         return apply_kfp(modify, self._cop, self)
 
-    def set_env(self, name, value):
+    def set_env(self, name, value=None, secret=None, secret_key=None):
+        """set pod environment var from value or secret"""
         i = 0
-        new_var = client.V1EnvVar(name=name, value=value)
+        value_from = None
+        if secret:
+            if value:
+                raise ValueError('cannot use both secret and value')
+            secret_key = secret_key or name
+            value_from = client.V1EnvVarSource(
+                secret_key_ref=client.V1SecretKeySelector(name=secret, key=secret_key)
+            )
+        else:
+            if value is None:
+                raise ValueError('value cannot be None')
+        new_var = client.V1EnvVar(name=name, value=value, value_from=value_from)
```

Comment:
Would be nicer to do validations in one place at the top, something like:
```
if (secret and value) or (not secret and value is None):
    raise ValueError('bla bla')
```

Reply:
Though I would just separate it to two different functions - `set_env` and `set_env_from_secret`
that way the function signature will be short and clear, right now it's already somewhat not-trivial (can't give both args), and later we'll want to add option to specify env var from configmap so it will become:
`def set_env(self, name, value=None, secret=None, secret_key=None, configmap=None, configmap_key=None):` ... the signature will just become bigger and bigger